sql-plugin/src/main/scala/com/nvidia/spark/rapids/PrioritySemaphore.scala

```diff
@@ -77,8 +88,10 @@ class PrioritySemaphore[T](val maxPermits: Int)(implicit ordering: Ordering[T])
     try {
       occupiedSlots -= numPermits
```

Comment:
I think there's a bug here where we can remove the permits, signal the next thread in the queue, but another thread flies in, grabs the lock, and sees that it has enough permits to acquire and does so.  In other words, it effectively jumped the priority queue.  To avoid this, we can have release perform the acquire on behalf of the threads it is waking up to remove this race (example code in another comment).